datafusion/functions-aggregate/src/count.rs

```diff
@@ -359,6 +547,15 @@ impl AggregateUDFImpl for Count {
     }
 }
 
+fn is_count_wildcard(args: &[Expr]) -> bool {
+    match args {
+        [] => true, // count()
+        // All const should be coerced to int64 or rejected by the signature
+        [Expr::Literal(ScalarValue::Int64(_))] => true, // count(1)
```

Comment:
We might need to consider `select count(null::bigint)`

Reply:
count(null) is not wildcard

```
D create table t(a int);
D insert into t values (1);
D select count(null) from t;
┌─────────────┐
│ count(NULL) │
│    int64    │
├─────────────┤
│           0 │
└─────────────┘
```

Reply:
So `is_count_wildcard()` should reject it.
It only has a problem in the display of logical plan.
![CleanShot 2025-02-20 at 11 10 02@2x](https://github.com/user-attachments/assets/2a6894a2-d5bc-47fc-a170-9dcfefbd2fd2)